systemFlake.nix

```diff
@@ -5,14 +5,19 @@
 , supportedSystems ? flake-utils-plus.lib.defaultSystems
 , inputs
 
-, nixosConfigurations ? { }
 , sharedExtraArgs ? { }
-, nixosProfiles ? { }
 , channels ? { }
 , channelsConfig ? { }
 , sharedModules ? [ ]
 , sharedOverlays ? [ ]
 
+, nixosConfigurations ? { }
+  # nixosProfiles should be deprecated since profiles aren't nixos specific
+, nixosProfiles ? { }
+, configProfiles ? nixosProfiles
+, defaultConfigOutput ? "nixosConfigurations"
```

Comment:
```suggestion
, defaultConfigType ? "nixosConfigurations"
```

Reply:
The whole `ouput` string is actually duplicating information that we should the builder to produce by itself.

I cannot think of a case where there is not a 1:1 mapping between builder and it's output key.

So this input can be spared from the public facing api.

Reply:
While that may be true, using the method I said, make the output of the builder '{ nixosConfigurations = }` is much harder to implement and adds complexity for the user trying to use it.
Its much easier to use to go
```nix
defaultConfigBuilder = inputs.darwin.lib.darwinSystem;
defaultConfigOutput = "darwinConfigurations;
```
And reading someone's flake.nix is much easier with this method.

Reply:
```nix
defaultConfigBuilder = args: { darwinConfigurations = inputs.darwin.lib.darwinSystem };
```
Is very complex for the user, for us, and the reader of flake.nix.

Reply:
Keeping output vs builder separation but dropped defaults, prefer #19.